server/src/common/utils/asyncUtils.ts

```diff
@@ -3,3 +3,5 @@ export const asyncForEach = async (array, callback) => {
     await callback(array[index], index, array)
   }
 }
+
+export const delay = (ms) => new Promise((resolve) => setTimeout(resolve, ms))
```

Comment:
Node v18 handle async setTimeout is you want:
https://nodejs.org/docs/latest-v18.x/api/timers.html#settimeoutcallback-delay-args

Reply:
good to know 💯 , but still experimental, we'll keep the utils for now